lib/alchemy-pg_search.rb

```diff
@@ -57,12 +58,16 @@ def self.index_page(page)
     # @return [ActiveRecord::Relation]
     def self.search(query, ability: nil)
       query = ::PgSearch.multisearch(query)
-                        .select("JSON_AGG(content) as content", :page_id)
-                        .reorder("")
-                        .group(:page_id)
-                        .joins(:page)
 
-      query = query.merge(Alchemy::Page.accessible_by(ability, :read)) if ability
+      if ability
+        # left_joins method is not usable here, because the order of the joins are incorrect
+        # and would result in a SQL error. We can receive the correct query order with these
+        # odd left join string
+        # Ref: https://guides.rubyonrails.org/active_record_querying.html#using-a-string-sql-fragment
+        query = query
+                  .joins("LEFT JOIN alchemy_pages ON alchemy_pages.id = pg_search_documents.page_id")
```

Comment:
Do we need the `left_join` outside of the `if` statement?

Reply:
It would make the query slower and Rails would reload the page again.